core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -304,6 +247,59 @@ class ReplicaFetcherThread(name: String,
   }
 
   /**
+    * - Truncate the log to the leader's offset for each partition's epoch.
+    * - If the leader's offset is greater, we stick with the Log End Offset
+    *   otherwise we truncate to the leaders offset.
+    * - If the leader replied with undefined epoch offset we must use the high watermark
+    */
+  override def maybeTruncate(fetchedEpochs: Map[TopicPartition, EpochEndOffset]): Map[TopicPartition, Long] = {
+    val truncationPoints = scala.collection.mutable.HashMap.empty[TopicPartition, Long]
+
+    fetchedEpochs.foreach { case (tp, epochOffset) =>
+      val replica = replicaMgr.getReplica(tp).get
+
+      if (epochOffset.hasError) {
+        info(s"Retrying OffsetsForLeaderEpoch fetch for epoch: $epochOffset as the leader reported an error: ${epochOffset.error}")
```

Comment:
This probably can be done in a followup patch. If there is an error, we probably want to add a bit of delay to the partition before retrying the OffsetsForLeaderEpoch request (like what we do when the fetch request has an error).